src/pkg/archive/archive_test.go

```diff
@@ -37,6 +37,83 @@ func readFile(t *testing.T, path string) string {
 	return string(data)
 }
 
+func generateAndCleanupSources(t *testing.T, sources []string) []string {
+	t.Helper()
+	paths := make([]string, len(sources))
+	root := t.TempDir()
+	for i, src := range sources {
+		path := filepath.Join(root, src)
+		content := strings.Join([]string{"test-data", src}, "-")
+		writeFile(t, path, content)
+		t.Cleanup(func() {
+			err := os.Remove(path)
+			if err != nil {
+				t.Fatalf("failed to remove file %s: %v", path, err)
+			}
+		})
+		paths[i] = path
+	}
+	return paths
+}
+
+func TestCompress(t *testing.T) {
+	t.Parallel()
+
+	tt := []struct {
+		name      string
+		sources   []string
+		dest      string
+		expectErr bool
+	}{
+		{
+			name:      "errors when both sources and dest are empty",
+			expectErr: true,
+		},
+		{
+			name:      "errors when given a dest but empty sources",
+			dest:      "err-archive.tar.gz",
+			expectErr: true,
+		},
+		{
+			name:      "errors when given sources but empty dest",
```

Comment:
This doesn't hit the intended error path for code coverage for empty dest given the `filepath.Join()` used in the test logic below. 

A permutation of this is still a valid testcase nonetheless as it represents a destination being a directory rather than a target but we should find a better way to augment the error as "dest is a directory" is less clear than "the destination is not a valid archive" or similar?

Reply:
It's true that the test environment setup does mean we're not actually passing an empty string into the Compress function here. Just so we're not blocking a critical bug fix, how about I go ahead and comment that test case out and we can figure out a path forward for the empty destination case in a follow up PR?

Reply:
Reframed the problem slightly and updated+added some test cases. lmk what you think